guide/sharding/additional-information.md

```diff
@@ -75,50 +77,52 @@ You can access them later as usual via `process.argv`, which contains an array o
 
 ## Eval arguments
 
-There may come the point where you will want to pass functions or arguments from the outer scope into a `.broadcastEval()` call.
+There may come the point where you will want to pass arguments from the outer scope into a `.broadcastEval()` call.
 
 ```js
-client.shard.broadcastEval(`(${funcName})('${arg}')`);
-```
-
-In this small snippet, an entire function is passed to the eval. It needs to be encased in parenthesis; it will throw errors on its way there otherwise. Another set of parenthesis is required so that the function gets called. Finally, the passing of the argument itself, which slightly varies, depending on the type of argument you are passing. If it's a string, you must wrap it in quotes, or it will be interpreted as is and will throw a syntax error because it won't be a string by the time it gets there.
-
-Now, what if you wanted to call a function from *within* the client context? That is to say, you are not passing a function. It would look something like this:
+function funcName(client, { arg }) {
+	// ...
+}
 
-```js
-client.shard.broadcastEval(`this.${funcName}(${args});`);
+client.shard.broadcastEval(funcName, { context: { arg: 'arg' } });
 ```
 
-This would become `client.funcName(args)` once it gets through. This is handy if you, for example, have extended your client object with your class and wish to call some of its methods manually.
+In this small snippet, an argument is passed to the `funcName` function through the `context` option of the <DocsLink path="typedef/BroadcastEvalOptions">`BroadcastEvalOptions`</DocsLink>.
+The function will recieve the arguments as an object as the second parameter.
+
+::: warning
+The `context` option only accepts properties which are JSON-serializable.
+This means you can not pass complex data types in the context directly.
+:::
 
 ### Asynchronous functions
 
 There may be a time when you want to have your shard process an asynchronous function. Here's how you can do that!
 
 ```js
-client.shard.broadcastEval(`
-	let channel = this.channels.cache.get('id');
+client.shard.broadcastEval(() => {
```

Comment:
```suggestion
client.shard.broadcastEval(client => {
```
Changing these empty params to use `client` to make it a bit more consistent, even if the variables don't get used